### PR TITLE
SPINEDEM-2692 Patch Patient tests

### PR DESCRIPTION
**Pull request by pl-nhs**
## Summary

:sparkles: New tests for patch patient

- Implements all patch patient tests that were in the existing pytest bdd codebase
- All patch tests can be run against the sandbox or veit07

:sparkles: Sandbox update

- Karate sandbox now has state, so it can support the patch tests
- New patients added so we can run tests in parallel
- This means the pytest tests for the old sandbox are no longer relevant for the Karate sandbox - so the ADO job has been removed
- All search scenarios can be run against the sandbox now
- Added validation of auth header

:sparkles: Small update to pytest sandbox scenarios - adding second "usual" name is not supported, I changed the request payload to make it more realistic

:sparkles: Removes the asid-required test job. This job is totally flaky, it is an immense annoyance every time I try to raise a PR, there is a ticket in the backlog to review this

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.

### CODE REVIEW

File: karate-tests/src/test/java/mocks/sandbox/supporting-functions.js [hunk 1]
```diff
@@ -86,30 +85,74 @@ function isValidUUID (uuid) {
...
-// eslint-disable-next-line no-unused-vars
-function validateHeaders (request) {
-  const X_REQUEST_ID = 'X-Request-ID'
-
+function validateAuthHeader (request) {
   let valid = true
+  let diagnostics = ''
+  // Check if the Authorization header is present and correct
+  const authorization = request.header('Authorization')
+  if (!('authorization' in request.headers)) {
```
Comment (ehsanhussain-nhsd) [inline karate-tests/src/test/java/mocks/sandbox/supporting-functions.js:114]: Can we not check if the authorization const declared above is not null instead?

File: karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/removeErrors.feature [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  Scenario: Error: attempt to remove name object from patient without prior test
+    * def diagnostics = "Invalid update with error - removal '/name/1' is not immediately preceded by equivalent test - instead it is the first item"
+    * def expectedBody = read('classpath:mocks/stubs/errorResponses/INVALID_UPDATE.json')
+
+    * def requestHeaders = call read('classpath:patients/healthcareWorker/healthcare-worker-headers.js')
+    * configure headers = requestHeaders 
+    * header Content-Type = "application/json-patch+json"
+    * header If-Match = etag
+    * path 'Patient', nhsNumber
+    * request {"patches":[{"op":"remove","path":"/name/1"}]}
```
Comment (ehsanhussain-nhsd) [inline karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/removeErrors.feature:31]: name/1 doesn't exist in the first place at this point. Although I understand that's not the point of this particular test.

File: karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/replace.feature [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  Background:
+    * def utils = call read('classpath:helpers/utils.feature')
+    * def accessToken = karate.callSingle('classpath:patients/healthcareWorker/auth-redirect.feature').accessToken
+    * url baseURL
+
+  Scenario: Replace attribute of an object
+    # To replace the attribute of an object, you need to provide the id of the object you want to replace
+    # in a preceding operation. (This is simlar to, but different from removing an object - we call 
+    # "replace" instead of "test")
+    * def nhsNumber = karate.env == 'mock' ? '9000000009' : '5900056597'
```
Comment (ehsanhussain-nhsd) [inline karate-tests/src/test/java/patients/healthcareWorker/updatePatientDetails/replace.feature:13]: Can the process of setting the NHS number and retrieving it be called in the Background portion like you've done elsewhere?

File: karate-tests/src/test/java/mocks/sandbox/supporting-functions.js (hunk 1 above)
Comment (pl-nhs) [reply karate-tests/src/test/java/mocks/sandbox/supporting-functions.js]: Okay I've improved line 114 with the simpler:

`if (authorization === null) {`

File: karate-tests/src/test/java/mocks/stubs/errorResponses/RESOURCE_NOT_FOUND.json [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
             "code": "RESOURCE_NOT_FOUND",
-            "display": "Resource not found"
+            "display": "Resource not found",
+            "system": "https://fhir.nhs.uk/R4/CodeSystem/Spine-ErrorOrWarningCode",
+            "version": "1"
           }
         ]
-      }
+      },
+      "severity": "information"
```
Comment (ehsanhussain-nhsd) [inline karate-tests/src/test/java/mocks/stubs/errorResponses/RESOURCE_NOT_FOUND.json:15]: Is "information" accurate for this?

Comment (pl-nhs) [reply karate-tests/src/test/java/mocks/stubs/errorResponses/RESOURCE_NOT_FOUND.json:15]: that's what gets returned, it's not my code...

**Review by ehsanhussain-nhsd — APPROVED** [review]

**Pull request merged**